AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -190,26 +190,42 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
                 self.processSyncMutationToCloudResult(result, mutationEvent: mutationEvent, api: api)
         }
 
-        dispatchOutboxStatusEvent(isEmpty: false)
-        operationQueue.addOperation(syncMutationToCloudOperation)
-        stateMachine.notify(action: .enqueuedEvent)
+        let localModel: Model
+        do {
+            localModel = try mutationEvent.decodeModel()
+            let outboxMutationEnqueuedEvent = OutboxMutationEvent
+                .fromModelWithoutMetadata(modelName: mutationEvent.modelName,
+                                          model: localModel)
+
+            dispatchOutboxMutationEvent(eventName: HubPayload.EventName.DataStore.outboxMutationEnqueued,
+                                        outboxMutationEvent: outboxMutationEnqueuedEvent)
+            dispatchOutboxStatusEvent(isEmpty: false)
+            operationQueue.addOperation(syncMutationToCloudOperation)
+            stateMachine.notify(action: .enqueuedEvent)
+        } catch {
+            log.error("Couldn't decode local model")
+            return
```

Comment:
this looks dangerous, if it fails, we just stop. to be safer, you could treat the dispatching of events as a best-effort and continue to add operation and notify
```
do {
} catch {
  log.error("Failed to decode mutationEvent model for OutboxMutationEvent")
}

operationQueue.addOperation(syncMutationToCloudOperation)
stateMachine.notify(action: .enqeuedEvent)
```

Reply:
In this case, I do the `decode()` in `do {}` block and construct a `OutboxMutationEvent` and also emit `outboxMutationEnqueued`, but if the json is failed to decoded, I add the operation into the queue, doesn't `outboxMutationEnqueued` also to be emitted?